packages/wrangler/src/deploy/deploy.ts

```diff
@@ -220,6 +216,24 @@ export const validateRoutes = (routes: Route[], hasAssets: boolean) => {
 					.join(`\n\n`)
 		);
 	}
+
+	if (mountedAssetRoutes.length > 0 && assetsDir !== undefined) {
+		const relativeAssetsDir = path.relative(process.cwd(), assetsDir);
+
+		logger.once.warn(
+			`Warning: The following routes will attempt to serve Assets on a mounted path:\n${mountedAssetRoutes
+				.map((route) => {
+					const routeNoScheme = route.replace(/https?:\/\//g, "");
+					const assetPath = path.join(
+						relativeAssetsDir,
+						routeNoScheme.substring(routeNoScheme.indexOf("/"))
+					);
+					return `  • ${route} (Will match assets: ${assetPath})`;
+				})
+				.join("\n")}` +
+				"\n\nRequests not matching an asset will be forwarded to the Worker."
```

Comment:
Are we ignoring the not found behaviour config setting here?

Reply:
Not matching assets where there is a Worker always go to the Worker (which can then handle in env.ASSETS)
I'll need to add something for no-Worker, need to check how we even detect that today